spec/requests/api/v2/providers/courses/get_for_accredited_body_by_provider_spec.rb

```diff
@@ -0,0 +1,50 @@
+require "rails_helper"
+
+RSpec.describe "GET /api/v2/recruitment_cycles/:year/providers/\
+  <provider_code>/training_providers/:training_provider_code/courses" do
+  let(:organisation) { create(:organisation) }
+  let(:user)         { create(:user, organisations: [organisation]) }
+  let(:payload)      { { email: user.email } }
+  let(:token)        { build_jwt(:apiv2, payload: payload) }
+  let!(:credentials) do
+    ActionController::HttpAuthentication::Token.encode_credentials(token)
+  end
+
+  let(:recruitment_cycle) { find_or_create(:recruitment_cycle) }
+  let(:accredited_body) { create(:provider, :accredited_body, organisations: [organisation]) }
+  let(:training_provider) { create(:provider) }
+
+  let!(:course1) { create(:course, provider: training_provider, accrediting_provider: accredited_body) }
+  let!(:course2) { create(:course, provider: training_provider) }
+
+  def perform_request
+    get request_path, headers: { "HTTP_AUTHORIZATION" => credentials }
+  end
+
+  let(:request_path) { "/api/v2/recruitment_cycles/#{recruitment_cycle.year}/providers/#{accredited_body.provider_code}/training_providers/#{training_provider.provider_code}/courses" }
```

Comment:
```suggestion
  let(:request_path) { "/api/v2/recruitment_cycles/#{recruitment_cycle.year}/providers/#{provider.provider_code}/training_providers/#{training_provider.provider_code}/courses" }
```

given the endpoint is for `provider` it reads a bit better to say that a provider has training provider... then jumping to `accredited_body` 

really just naming of things

Reply:
Sorry, but I like the added context given by calling it `accredited_body` in this particular case.